RetailCoder.VBE/Inspections/QuickFixes/PassParameterByReferenceQuickFix.cs

```diff
@@ -19,21 +20,26 @@ public PassParameterByReferenceQuickFix(ParserRuleContext context, QualifiedSele
 
         public override void Fix()
         {
-            string pattern = "^\\s*" + Tokens.ByVal + "(\\s+)";
-            string rgxReplacement = Tokens.ByRef + "$1";
-            Regex rgx = new Regex(pattern);
+            var byValParameter = Context.GetText();
 
-            var parameter = Context.GetText();
-            var newParameter = rgx.Replace(parameter, rgxReplacement);
+            var byRefParameter = BuildByRefParameter(byValParameter);
 
-            var selection = Selection.Selection;
+            ReplaceByValParameterInModule(byValParameter, byRefParameter);
+        }
 
+        private string BuildByRefParameter(string originalParameter)
+        {
+            var everythingAfterTheByValToken = originalParameter.Substring(Tokens.ByVal.Length);
+            return Tokens.ByRef + everythingAfterTheByValToken;
+        }
+        private void ReplaceByValParameterInModule( string byValParameter, string byRefParameter)
+        {
+            var selection = Selection.Selection;
             var module = Selection.QualifiedName.Component.CodeModule;
-            {
-                var lines = module.GetLines(selection.StartLine, selection.LineCount);
-                var result = lines.Replace(parameter, newParameter);
-                module.ReplaceLine(selection.StartLine, result);
-            }
+
+            var lines = module.GetLines(selection.StartLine, selection.LineCount);
+            var result = lines.Replace(byValParameter, byRefParameter);
```

Comment:
The line relevant to using replace on a line of code is `bar = barTwo`. If it's only replacing `bar` with `xbar`, it can't change that line to `xbar = xbarTwo`, or it breaks it.